docs/ref/models/window.txt

```diff
@@ -0,0 +1,134 @@
+        >>>
+        >>> Movie.objects.prefetch_related('director').annotate(
+        >>>     avg_rating=Window(
+        >>>         expression=Avg('rating'),
+        >>>         partition_by=[F('genre'), F('country')],
+        >>>         order_by=[F('director__name').asc(), F('released__year')],
+        >>> ))
+
+    .. note::
+
+        Implementation and support varies widely among the different backends,
+        e.g., not everything in ``OrderBy`` may be supported. Please read the
+        documentation of your database before using this expression.
+
+    .. note::
+
+        The SQL standard disallows referencing the window function in the
+        ``WHERE``-clause and Django raises an exception if it's attempted.
+        This scenario may be solved with subqueries.
+
+.. _window-frames:
+
+Frames
+======
+
+Two classes are used to model a frame to distinguish between a range-based
+sequence of rows, and an ordinary sequence of rows. It's encouraged to read
+the documentation of your database of choice to determine if all features of
+the SQL standard are supported by your database.
+
+.. class:: ValueRange(start=None, end=None)
+
+    .. attribute:: frametype
+
+        This attribute is set to ``RANGE``.
+
+.. class:: RowRange(start=None, end=None)
+
+    .. attribute:: frametype
+
+        This attribute is set to ``ROWS``.
+
+Frames are used for narrowing which rows that should be used for computing the
+result. A peer in a frame is a row with an equivalent value, or all rows in
+case of an absent ordering-clause. The default starting point for a frame is
+``UNBOUNDED PRECEDING`` which is the first row of the partition, and the
+default end point is ``CURRENT ROW``. The end point is always explicitly
+included in the SQL generated by the ORM. ``UNBOUNDED FOLLOWING`` marks the
+end of the partition. The default frame includes all rows from the partition
+to the last row in the set.
+
+The accepted values for the parameters for ``start`` and ``end`` are either
+``None``, or an integer. The convention for the input values is as follows. A
+negative integer for ``start`` results in ``N preceding``, 0 results in
+``CURRENT ROW``, and None or a positive integer yield ``UNBOUNDED
+PRECEDING``. Positive integers are accepted for ``end``.
+
+There is a difference what ``CURRENT ROW`` includes, i.e., when specified in
+``ROWS`` mode, the frame starts or ends with the current row. When specified in
+``RANGE``, the frame starts or ends at the first or last peer according to the
+ordering-clause.
+
+   .. admonition:: Ordering and frames.
+
+      It is recommended to specify a unique ordering when using the ``ROWS``
+      option to avoid unpredictable results. For ``RANGE``, the ordering
+      ensures that identical peers are either in the frame, or not.
+
+      >>> from django.db.models import F, RowRange, Window
+      >>>
+      >>> Book.objects.prefetch_related('author').annotate(avg_rating=Window(
```

Comment:
Can you add a line break before `avg_rating=` please. I.e. follow the same indentation design as above.